File: VB.NET/SelectPdfPageForm.vb
Imports System.Collections
Imports System.Drawing
Imports System.Windows.Forms
Imports Vintasoft.Barcode

Public Partial Class SelectPdfPageForm
	Inherits Form

	#Region "Fields"

	Private _labelText As String
	Private _selected As Boolean

	#End Region



	#Region "Constructors"

	Public Sub New()
		InitializeComponent()
		_labelText = label1.Text
	End Sub

	#End Region



	#Region "Methods"

    Private Sub FormPdfPageSelect_FormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs) Handles MyBase.FormClosing
        Visible = False
        e.Cancel = True
    End Sub

    Private Sub selectButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles selectButton.Click
        _selected = True
        Close()
    End Sub


	Private Function GetImageFromPage(viewer As PdfImageViewer, pageIndex As Integer) As Image
		Dim names As String() = viewer.GetImageNames(pageIndex)
		Dim images As New ArrayList()
		For i As Integer = 0 To names.Length - 1
			Try
				images.Add(viewer.GetImage(pageIndex, names(i)))
			Catch e As Exception
				MessageBox.Show(String.Format("Image '{0}': {1}", names(i), e.Message), "Error", MessageBoxButtons.OK, MessageBoxIcon.[Error])
			End Try
		Next
		If images.Count = 0 Then
			Return Nothing
		End If
		If images.Count = 1 Then
			Return DirectCast(images(0), Image)
		End If

		' merge images
		Dim padding As Integer = 5
		Dim heigth As Integer = 0
		Dim width As Integer = 0
		Dim n As Integer = images.Count
		For i As Integer = 0 To n - 1
			Dim current As Image = DirectCast(images(i), Image)
			If width < current.Width Then
				width = current.Width
			End If
			heigth += current.Height
		Next
		width += 3
		heigth += (n + 1) * padding
		Dim result As New Bitmap(width, heigth, System.Drawing.Imaging.PixelFormat.Format24bppRgb)
		Dim g As Graphics = Graphics.FromImage(result)
		g.FillRectangle(Brushes.White, New Rectangle(0, 0, width, heigth))
		Dim dy As Integer = 1
		For i As Integer = 0 To n - 1
			Dim current As Image = DirectCast(images(i), Image)
			g.DrawImageUnscaled(current, New Point(1, dy))
			dy += current.Height + padding
			current.Dispose()
		Next
		g.Dispose()

		Return result
	End Function

	Public Function SelectImage(pdfFileName As String) As Image
		Dim viewer As PdfImageViewer
		Try
			viewer = New PdfImageViewer(pdfFileName)
		Catch ex As Exception
			MessageBox.Show(ex.Message, "Error", MessageBoxButtons.OK, MessageBoxIcon.[Error])
			Return Nothing
		End Try

		Try
			pages.Items.Clear()
			For i As Integer = 0 To viewer.PageCount - 1
				Dim k As Integer = viewer.GetImageNames(i).Length
				If k > 0 Then
					pages.Items.Add(i + 1)
				End If
			Next
			If pages.Items.Count = 0 Then
				MessageBox.Show("Images in PDF file are not found.", "Error", MessageBoxButtons.OK, MessageBoxIcon.[Error])
				Return Nothing
			End If
			pages.SelectedIndex = 0

			If pages.Items.Count = 1 Then
				Return GetImageFromPage(viewer, 0)
			End If

			label1.Text = String.Format(_labelText, pages.Items.Count)

			_selected = False
			ShowDialog()
			If _selected Then
				Dim result As Image = GetImageFromPage(viewer, CInt(pages.SelectedItem) - 1)
				Return result
			End If

			Return Nothing
		Finally
			viewer.Dispose()
		End Try
	End Function

	#End Region

End Class

File: CSharp/SelectPdfPageForm.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Drawing;
using System.Windows.Forms;

using Vintasoft.Barcode;
using Vintasoft.Imaging;

namespace BarcodeDemo
{
    /// <summary>
    /// A form that allows to select a page in PDF document.
    /// </summary>
    public partial class SelectPdfPageForm : Form
    {

        #region Fields

        /// <summary>
        /// The format string of information label.
        /// </summary>
        string _labelFromatString;

        /// <summary>
        /// Indicates that PDF page can be selected.
        /// </summary>
        bool _pageSelected;

        #endregion



        #region Constructors

        /// <summary>
        /// Initializes a new instance of the <see cref="SelectPdfPageForm"/> class.
        /// </summary>
        public SelectPdfPageForm()
        {
            InitializeComponent();
            _labelFromatString = label1.Text;
        }

        #endregion



        #region Methods

        /// <summary>
        /// Handles the FormClosing event of this form.
        /// </summary>
        private void FormPdfPageSelect_FormClosing(object sender, FormClosingEventArgs e)
        {
            Visible = false;
            e.Cancel = true;
        }

        /// <summary>
        /// Closes this form.
        /// </summary>
        private void selectButton_Click(object sender, EventArgs e)
        {
            _pageSelected = true;
            Close();
        }


        /// <summary>
        /// Gets the image from PDF page with specified index.
        /// </summary>
        /// <param name="viewer">The PDF viewer.</param>
        /// <param name="pageIndex">Index of the page.</param>
        private Image GetImageFromPage(PdfImageViewer viewer, int pageIndex)
        {
            string[] names = viewer.GetImageNames(pageIndex);
            List<VintasoftBitmap> images = new List<VintasoftBitmap>();
            for (int i = 0; i < names.Length; i++)
            {
                try
                {
                    images.Add(viewer.GetImage(pageIndex, names[i]));
                }
                catch (Exception e)
                {
                    MessageBox.Show(string.Format("Image '{0}': {1}", names[i], e.Message), "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                }
            }
            if (images.Count == 0)
                return null;
            if (images.Count == 1)
                return GdiConverter.Convert(images[0], true);

            // merge images
            int padding = 5;
            int heigth = 0;
            int width = 0;
            int n = images.Count;
            for (int i = 0; i < n; i++)
            {
                VintasoftBitmap current = images[i];
                if (width < current.Width)
                    width = current.Width;
                heigth += current.Height;
            }
            width += 3;
            heigth += (n + 1) * padding;
            Bitmap result = new Bitmap(width, heigth, System.Drawing.Imaging.PixelFormat.Format24bppRgb);
            using (Graphics g = Graphics.FromImage(result))
            {
                g.FillRectangle(Brushes.White, new Rectangle(0, 0, width, heigth));
                int y = 1;
                for (int i = 0; i < n; i++)
                {
                    using (Bitmap current = GdiConverter.Convert(images[i], true))
                    {
                        g.DrawImageUnscaled(current, new Point(1, y));
                        y += current.Height + padding;
                    }
                }
            }

            return result;
        }

        /// <summary>
        /// Selects the image from PDF document.
        /// </summary>
        /// <param name="pdfFileName">Name of the PDF file.</param>
        public Image SelectImage(string pdfFileName)
        {
            PdfImageViewer viewer;
            try
            {
                viewer = new PdfImageViewer(pdfFileName);
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                return null;
            }

            try
            {
                pages.Items.Clear();
                for (int i = 0; i < viewer.PageCount; i++)
                {
                    int k = viewer.GetImageNames(i).Length;
                    if (k > 0)
                        pages.Items.Add(i + 1);
                }
                if (pages.Items.Count == 0)
                {
                    MessageBox.Show("Images in PDF file are not found.", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                    return null;
                }
                pages.SelectedIndex = 0;

                if (pages.Items.Count == 1)
                {
                    return GetImageFromPage(viewer, 0);
                }

                label1.Text = string.Format(_labelFromatString, pages.Items.Count);

                _pageSelected = false;
                ShowDialog();
                if (_pageSelected)
                {
                    Image result = GetImageFromPage(viewer, (int)pages.SelectedItem - 1);
                    return result;
                }

                return null;
            }
            finally
            {
                viewer.Dispose();
            }
        }

        #endregion

    }
}